scripts/docs_link_checker.py

```diff
@@ -0,0 +1,306 @@
+import glob
+import logging
+import os
+import re
+from typing import List
+
+import click
+import requests
+
+logger = logging.getLogger()
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+class LinkReport:
+    def __init__(self, link: str, file: str, message: str):
+        self.link = link
+        self.file = file
+        self.message = message
+
+    def __str__(self):
+        return f"{self.message}: File: {self.file}, Link: {self.link}"
+
+
+class LinkChecker:
+    def __init__(
```

Comment:
If the lift isn't particularly big, would you mind adding docstrings to these classes? Even a single line will do!

Reply:
Actually in hindsight, this is a simple tool and doesn't require such a high level of documentation. Feel free to disregard.

Reply:
done!